lib/logic/auth/facebook/facebook_sign_client.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter_facebook_login/flutter_facebook_login.dart';
+
+enum SingAnswer { successfully, cancelled, error, logOut }
+
+class FacebookSignClient {
+
+  FacebookSignClient() {
+    _facebookLogin = FacebookLogin();
+  }
+
+  FacebookLogin _facebookLogin;
+
+  Future<SingAnswer> handleSignOut() async {
+    await _facebookLogin.logOut();
+    return SingAnswer.logOut;
+  }
+
+  Future<SingAnswer> handleSignIn() async {
+
+    _facebookLogin.loginBehavior = FacebookLoginBehavior.webViewOnly;
+    final FacebookLoginResult result = await _facebookLogin.logIn(<String>
+      ['email']);
+    SingAnswer loginAnswer = SingAnswer.error;
+
+    switch (result.status) {
+      case FacebookLoginStatus.loggedIn:
+        // send token(result.accessToken) to backend.
+        loginAnswer = SingAnswer.successfully;
+        break;
+      case FacebookLoginStatus.cancelledByUser:
+        loginAnswer = SingAnswer.cancelled;
+        break;
+      case FacebookLoginStatus.error:
+        loginAnswer = SingAnswer.error;
+        break;
+    }
+    return loginAnswer;
```

Comment:
The idea is good but we need to think carefully if this is the best approach. I guess this is an attemp to achieve an status independent from the FB library. We still need to know if this status make sense in other SignIn libraries. Even if we already knew that, this is not the PR to do that. We should do this in a refactor process once the libraries are integrated. We need to collect information before starting an abstraction process.